app/models/conditions_response/user_email_alert.rb

```diff
@@ -0,0 +1,116 @@
+# Abstract parent class for all alerts that send out emails to users
+#
+#  * knows if an alert needs to be sent today (or on _this_date_)
+#  * knows the name of the method to send to MemberMailer to send out the
+#     alert to a user
+#
+#  TODO If each class is responsible for doing 1 thing -- i.e. sending out 1
+#   kind of alert for 1 condition, then is 'condition' really needed?
+#     (If/when we have situations when 1 class needs to handle more than 1
+#      condition _and_ we must have it handled only within 1 class,
+#        then we can create the condition attribute.)
+#
+class UserEmailAlert
+
+
+  # TODO this class is an instantiation of the Condition; why pass it again as an arg?
+  def self.condition_response(_condition, config, log, this_date: DateTime.now.utc)
```

Comment:
OK.  I see how you have this working.  My issue (and initial confusion) is that the Condition class and 'Alert' classes (_= the classes that process_conditions_) have too much coupling.  You're making the Alert classes dependent on the Condition class.

The information an Alert class needs to do its processing should be passed in to it.  It shouldn't have  know how to dig it out of a class to get to it.

in process_conditions.rake this line:
```
klass.condition_response(condition, log)
```

should be changed to:
```
klass.condition_response(condition.config, condition.timing, log)
```

thus an Alert class needs to know nothing about a Configuration class and so does not have coupling to it. Less coupling = less brittle.

Reply:
Perhaps .... but I think of passing the `condition` instance as not increasing coupling but being more flexible.   The receive of the argument does not know the internals of `condition`, but it does know messages the the instance will respond to - it uses these to get the data it needs.  Also, if we later decide that we need another `condition` attribute to be passed to a response class (sorry to change terminology here, but I'm still trying to figure out how best to _name things_), then we add that to the API - that strikes me as _more_ brittle.

The design is such that the author of a response class _has_ to know how to access attributes of a `condition` instance - that includes, but is not limited to, the design of the `config` data (which probably won't be limited to a "standard" `days` array).

Reply:
Hm.  Ok.  I think we're converging on the same stuff:  what information does an `Alert` need in order to do what it needs to do?  I'm thinking that it all goes under the `configuration` concept -- and agree that we don't yet know all that it might entail.
I _think_ that you're seeing it as some goes under the `configuration` concept, and some might go more generally under the `condition` concept.  
Is that right?  Or am I still missing the mark?

(and speaking of _naming:_  I keep using the term `Alerts` but that's just a placeholder for whatever those classes are.  For now I am only working with alerts, so that fits.  But again -- it's just a placeholder)

Reply:
"and some might go more generally under the condition concept.
Is that right? " - yes, exactly how I'm thinking of it.